beps/0006-docs-personas-framework-portal/README.md

```diff
@@ -0,0 +1,240 @@
+
+## Motivation
+
+The current Backstage documentation has been reported to be difficult to navigate, making it challenging for different personas across the DevEx ecosystem to extract value. The CNCF [**assessment**](https://github.com/cncf/techdocs/tree/main/assessments/0008-backstage/) and the resulting [**issues**](https://github.com/backstage/backstage/issues/21893) highlight the need for improvement in the documentation structure.
+
+### Goals
+
+- Divide the documentation into two section: Framework and Developer Portal
+- Define the personas Backstage is targeting
+- Structure the documentation to cater the different personas
+- Move the existing content to the appropriate section
+
+### Non-Goals
+
+- Rewrite the entire documentation from scratch
+- Write additional content beyond the scope of the existing documentation
+
+## Proposal
+
+The proposed restructuring of the Backstage documentation revolves around two core ideas:
+
+1. Positioning Backstage as both a framework to build developer portals and a developer portal itself, and splitting the documentation into two main sections:
+
+   - Developer Portal: Focusing on the features, configuration, and usage of the developer portal that users get out of the box with the core plugins.
+   - Framework: Covering the aspects of Backstage as a framework, including guides for integrators and builders who want to create their own developer portal using Backstage.
+
+2. Defining the personas participating in Backstage adoption journeys to improve documentation navigation. The identified personas are:
+
+   - **User**: A person who uses Backstage to find information, use plugins, and consume the developer portal.
+   - **Administrator/Operator**: A person who configures, secures, and deploys the developer portal, manages plugins, and oversees the general administration of the developer portal.
+   - **Integrator/Builder**: A person who builds plugins, customizes the code and design, and creates custom-built developer portals based on the Backstage framework. This includes developers and designers and anyone adding new functionality to their own Backstage instance.
+   - **Product Manager/Business stakeholders**: A person who defines the strategy for adopting Backstage, identifies use cases, communicates the value proposition for adopting Backstage and connects the developer portal to the business strategy.
+   - **Contributor**: A person who contributes to the Backstage upstream ecosystem.
+
+The adoption strategy would be as follows:
+
+- Create a dedicated page describing the personas Backstage is targeting and the documentation sections that cater to each persona.
+- Restructure the documentation into the two new sections (Framework and Developer Portal) and redistribute the existing content accordingly.
```

Comment:
What are we scoping as "restructuring" or "redistributing"? Is this primarily just moving the existing docs into the separate ToCs? While working on #22541, I noticed a lot of documentation that is either out of date, doesn't have a great narrative within or doesn't fit into the overall narrative for the section.

Reply:
I think it is a mix of things, but I found it too hard to describe it in detail in a single BEP. But more than open to any suggestions.

For this one specifically, I thought about the following:

1. Aligning the concepts related to Personas, Backstage Developer Portal, and Framework to Build Developer Portals.
2. Dividing the Docs in two sections: Developer Portal and Framework
3. Updating overviews and descriptions according to the proposed changes.
4. Updating the table of contents, taking into considerations the proposed Personas and the CNCF [recommendation](https://github.com/backstage/backstage/issues/21946) 
5. Redistributing content according to the new structure

Next steps:

6. Completing the updates based on the [CNCF ](https://github.com/backstage/backstage/issues/21893)recommendations
7. Renaming sections to standardizing the content
8. Review each individual page to update and correct content

Reply:
I'm a bit torn here, I like the idea of separating the documentation into the two separate sections you describe, but I don't think changing the ToC is where we should start, especially if outputs of this update are going to be an announcement of the changes. IMO, we should align on narrative throughout the documentation and general parameters for how to write documentation for specific user profiles first before diving in to updating the organization. That would give us a shared understanding of what each role means concretely and lay the foundation for organization or content updates.

Priorities in my mind:
1. Create a documentation narrative overview
2. Create guidelines for writing docs for Admins/Devs/Integrators
3. Start condensing/adjusting existing documentation to the guidelines above and relocate in the existing ToC.
4. Decide where to split the documentation and do it in a single PR or small set of PRs.
5. Announce.

I'm definitely out of touch with the current adopter experience, but I don't quite understand the focus on differentiating framework and developer portal rn, that feels like something nuanced enough that we should tackle it specifically in the relevant sections -- most likely overview.

tldr; I think we should start from the bottom-up instead of top-down, starting with a set of guidelines for narrative and voices for each of the roles, then working to update the current table of contents organically.

Reply:
I follow you @aramissennyeydd, and I think I understand your perspective. 

Defining a clear top-down structure is, in my opinion, crucial before making any significant changes to the documentation. Without a well-defined structure, it can be confusing and challenging to determine what changes to make and how to comfortably relocate content. 

Although I initially proposed separating the docs into two products – a Developer Portal and a Framework to Build Developer Portals – a similar outcome could be accomplished by reorganizing the Table of Contents (ToC) as described by the CNCF [**recommendation**](https://github.com/backstage/backstage/issues/21946). Following a similar approach, WordPress clearly separates its [**general documentation**](https://wordpress.org/documentation/) from its [**framework documentation**](https://developer.wordpress.org/).

To clarify, the proposal's intention is to define the personas involved in the Backstage adoption journey and to remove adoption barriers by differentiating the content targeting developers, so that new adopters, assessing, trying, deploying, and using Backstage can follow content that doesn't require deep knowledge of the Backstage codebase to understand what it is, what problem it solves, and can start realizing value from it. This could be accomplished by separating the docs into "two products" or by separating the Developer/Integrator content into its own section, as [**proposed**](https://github.com/backstage/backstage/issues/21946) by the CNCF.

I believe clearly defining the user personas, outlining the structure of the ToC, and relocating content accordingly is a critical step to identify the gaps, given that the current docs have mixed content for admins, developers/integrators, users, and business guides while heavily focused on the Developer persona. This is, IMO why most of the [**general recommendations**](https://github.com/backstage/backstage/issues/21893) are related to creating Admin guides. However, plenty of content exists related to Admin guides in the existing docs but is mixed with Developer-oriented content, which is why restructuring the ToC would guide us on what content to move where. Just my two cents.

Reply:
Alright, I'm following, I think we're talking in pretty similar directions. I'd like to add a more prescriptive description of each of the user roles and each high level ToC item, that's the missing piece for me right now. Relocating and updating the documentation in flight makes sense, but I'd like to know more holistically why we're putting certain docs in certain places. I'm at BackstageCon so it'll be a little bit of turnaround, might not be able to get to this until tonight.

Reply:
Comparing across forks on GH is a little weird, ended up opening https://github.com/waldirmontoya25/backstage/pull/1 with possible updates.